add CarrierConfig override UI

inthewaves opened the pull request:
Closes https://github.com/GrapheneOS/os-issue-tracker/issues/956

Requires changes in
- frameworks/base to fix some issues with UI components, expose a method to return config overrides in the CarrierConfigLoader, and infrastructure for extended SIM state: https://github.com/GrapheneOS/platform_frameworks_base/pull/256
- frameworks/opt/telephony: https://github.com/GrapheneOS/platform_frameworks_opt_telephony/pull/11
- platform/packages/TelephonyProvider for infrastructure for extended SIM state (https://github.com/GrapheneOS/platform_packages_providers_TelephonyProvider/pull/3)
- https://github.com/GrapheneOS/platform_packages_apps_CarrierConfig2/pull/4
- packages/app/Settings (here)
- packages/services/Telephony for implementing the method in CarrierConfigLoader to return config overrides: https://github.com/GrapheneOS/platform_packages_services_Telephony/pull/18

To add more options, see CarrierConfigOverrideOptions.kt.

<img src="https://github.com/user-attachments/assets/e3e1cc8a-c803-48e1-978a-897572470421" width=35% />

inthewaves committed:
add CarrierConfig override UI

Logic for overriding configs is handled in other repos

To add more options, see CarrierConfigOverrideOptions.kt.

Part of changeset with changes to
- frameworks/base
- frameworks/opt/telephony
- packages/providers/TelephonyProvider
- packages/apps/CarrierConfig2

Fei1Yang commented:
May I also request VoWiFi overrides to be added too? it's also very useful to workaround bad base station coverage or to prevent being charged with roaming rates.

inthewaves committed:
CarrierConfig override: warn about overrides from other sources

User-facing overrides have only been implemented via CarrierConfig2. Display a warning if there are
overrides present in AOSP's CarrierConfigLoader.

Requires changes to packages/services/Telephony and frameworks/base for the new method in
ICarrierConfigLoader.aidl to return override configs explicitly.
